src/persistence/keyvalue/NoStorageMapKeyValueStore.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { KeyValueHandler } from "./KeyValueHandler";
+
+export class NoStorageMapKeyValueStore implements KeyValueHandler {
+    protected data = new Map<string, any>();
+
+    constructor() {}
+
+    private buildKeyString(context: string, contextKey: string): string {
+        return `${context}$$${contextKey}`;
+    }
+
+    getContextKey(context: string, contextKey: string, defaultValue: any): any | undefined {
+        const value = this.data.get(this.buildKeyString(context, contextKey));
+        if (value === undefined) {
+            return defaultValue;
+        }
+        return value;
+    }
+
+    getContextKeys(context: string): { key: string, value: any}[] {
+        const keys = Array.from(this.data.keys());
+        return keys.filter((key) => key.startsWith(`${context}$$`)).map((key) => ({ key: key.substring(context.length + 1), value: this.data.get(key)}));
```

Comment:
I'd iterate over the map itself (gives you entries):
```suggestion
return [...this.data]
    .filter(([key]) => key.startsWith(`${context}$$`))
    .map(([key, value]) => ({ key: key.substring(context.length + 1), value }));
```

Reply:
Good idea, thank you!